scalyr_agent/builtin_monitors/windows_event_log_monitor.py

```diff
@@ -823,6 +936,47 @@ def _strip_xmltodict_prefixes(x):
         return x
 
 
+class _DLL:
+    @staticmethod
+    def dllpath(channel, provider):
+        keyname = "SYSTEM\\CurrentControlSet\\Services\\EventLog\\%s\\%s" % (
+            channel,
+            provider,
+        )
+        keyhandle = win32api.RegOpenKey(win32con.HKEY_LOCAL_MACHINE, keyname)
+        try:
+            return win32api.ExpandEnvironmentStrings(
+                win32api.RegQueryValueEx(keyhandle, "ParameterMessageFile")[0]
+            )
+        finally:
+            win32api.RegCloseKey(keyhandle)
+
+    def __init__(self, channel, provider):
+        try:
+            self.path = self.dllpath(channel, provider)
+            self.handle = win32api.LoadLibraryEx(
+                self.path, 0, win32con.LOAD_LIBRARY_AS_DATAFILE
+            )
+        except Exception as e:
+            self.handle = None
+            raise e
+
+    def __del__(self):
```

Comment:
IIRC, ``__del__`` is Python implementation and GC algorithm specific so it may be safer / make more sense to use a context manager or similar (although not sure, how hard would it be to re-organize the code to use context manager or similar instead).

Reply:
Yeah I wasn't happy with this either, will revisit it